### ISSUE TITLE
[doc] 11.1  Do’s and Don’ts

### ISSUE THREAD

**Issue by rbeldin** [SUSE/suse-best-practices#90]
11.1  Do’s and Don’ts:

https://documentation.suse.com/sbp/all/html/SLES4SAP-hana-sr-guide-PerfOpt-12/index.html#_dos_and_donts

I think the 4th bullet is really part of the 3rd bullet.  Am I correct?   See below.    The language is also a little awkward, perhaps:  Adding location rules for the clone, master/slave, or IP resource.  Only location rules mentioned in this setup guide are allowed as 'migrating' or 'moving' resources in crm-shell, HAWK, or other tools would add client-prefer rules that would break the configuration developed in this document.  

In your project, avoid:

Rapidly changing/changing back cluster configuration, such as: Setting nodes to standby and online again or stopping/starting the master/slave resource.

Creating a cluster without proper time synchronization or unstable name resolutions for hosts, users and groups

Adding location rules for the clone, master/slave or IP resource. Only location rules mentioned in this setup guide are allowed.

As "migrating" or "moving" resources in crm-shell, HAWK or other tools would add client-prefer location rules this activities are completely forbidden.

**Comment by chabowski** [comment]
Hi @rbeldin, thanks very much for your comment! I will discuss it with the author of the doc and have the content changed accordingly. Really appreciated! Kind regards - Meike

**Comment by fmherschel** [comment]
As far as I remember Lars already answered the questions. Is there still something we did not answer?

**Comment by fmherschel** [comment]
@rbeldin: You are right both items technically handle about constraints, ban-rules and so on. The difference is, that the first is an explicit add of rules while the second is adding rules implicitly. Customers often have overseen that "migrating" resources does add either ban rules (--force) or location rules (if positive). This is why we separated that. They just do not know that a simple klick in HAWK (as example) does change the configuration of the cluster.

**Comment by lpinne** [comment]
@rbeidin: As fmherschel wrote, many customers do not know the consequences of moving resources by HAWK or crm.
Left-over migration constraints are among the favourite causes for support requests. 
The next version of the guide will contain something like:
- As "migrating" or "moving" resources in crm-shell, HAWK or other tools would add client-
prefer location rules, support is limited to maintenance procedures described in this doc-
ument. See Section “Testing the Cluster” and Section “Maintenance” for proven proce-
dures.

**Comment by rbeldin** [comment]
That sounds very good.   We recently dealt with a customer who had a bunch of constraint issues left over from when they were migrating some of the cloned resources and it took quite some time to resolve.  Thanks for the quick response!   I think we can close this but I will leave the actual closing up to you in case you need to wrap this up in some way.

**Comment by fmherschel** [comment]
Thanks for your feedback! We will mark that section as "todo" in the sources so we do not forget to update it. Later we will close the issue.

**Comment by lpinne** [comment]
see setup guides
